File: notes-springboot/src/main/java/com/spring/notes/services/NoteService.java
package com.spring.notes.services;

import com.spring.notes.entities.Note;

import java.util.List;

public interface NoteService {
    Note getNote(long noteId);
    void addNote(String username, Note note);
    void deleteNote(long noteId);
    List<Note> listNotes(String username);
}

File: notes-reactjs/src/notes/api/NoteService.js
import axios from 'axios';

const BASE_URL = "http://localhost:8090"

function addNote(note) {
    return axios.post(`${BASE_URL}/notebooks/1/notes`, note)
}

function deleteNote(noteId) {
    return axios.delete(`${BASE_URL}/notes/${noteId}`);
}

function listNotes() {
    return axios.get(`${BASE_URL}/notebooks/1/notes`);
}

export {
    BASE_URL, addNote, deleteNote, listNotes
};